lib/linter-registry.coffee

```diff
@@ -0,0 +1,68 @@
+{Emitter} = require('atom')
+validate = require('./validate')
+helpers = require('./helpers')
+
+class LinterRegistry
+  constructor: ->
+    @linters = []
+    @locks =
+      Regular: new WeakSet
+      Fly: new WeakSet
+    @emitter = new Emitter
+
+  getLinters: ->
+    return @linters.slice() # Clone the array
+
+  hasLinter: (linter) ->
+    @linters.indexOf(linter) isnt -1
+
+  addLinter: (linter) ->
+    try
+      validate.linter(linter)
+      @linters.push(linter)
+    catch e then helpers.error(e)
+
+  deleteLinter: (linter) ->
+    return unless @hasLinter(linter)
+    @linters.splice(@linters.indexOf(linter), 1)
+
+  lint: ({onChange, editorLinter}) ->
+    editor = editorLinter.editor
+    lockKey = if onChange then 'Fly' else 'Regular'
+    return if onChange and not atom.config.get('linter.lintOnFly')
+    return unless editor is atom.workspace.getActiveTextEditor()
+    return unless editor.getPath()
+    return if @locks[lockKey].has(editorLinter)
+
+    @locks[lockKey].add(editorLinter)
+    scopes = editor.scopeDescriptorForBufferPosition(editor.getCursorBufferPosition()).scopes
+    scopes.push('*') # To allow global linters
+
+    # Confusing code ahead, proceed with caution :P
```

Comment:
Also comment is confusing as it just states danger, does not help understand code.   Remove it.